### ISSUE TITLE
"make test" doesn't explicitly test SSL connections

### ISSUE THREAD

**Issue by esabol** [gearman/gearmand#56]
Yes, there are SSL tests in the test suite, but they're part of the existing tests. It would be nice if there was an _explicit_ "ssl" test that would SKIP if the compilation did not have "SSL" enabled and "PASS" if SSL connections are working.

I spent a couple hours yesterday trying to debug out why my SSL connections weren't working before I realized that SSL wasn't enabled (even though I compiled with "--enable-ssl"). It turns out my Ubuntu Docker container did not have the libssl-dev package installed, so "--enable-ssl" failed to actually enable SSL. If there was an explicit SSL test in the test suite, I would have noticed it a lot quicker.

**Comment by p-alik** [comment]
SSL testing independently I guess `configure` should fail if some build requirements are not present.
But indeed there is highly demand for SSL testing.

**Comment by SpamapS** [comment]
Agreed, if I ask for --with-ssl I'd expect it to explode if I can't build with SSL.

That's a good candidate for a separate issue, as this one is more about the explicit SSL testing.

**Comment by esabol** [comment]
Yeah, that makes complete sense. I've created a separate issue #57 for that.

Do you think there's sufficient coverage of SSL connections in the current test suite?
